src/dune_pkg/rev_store.ml

```diff
@@ -32,8 +32,7 @@ module Remote = struct
   type nonrec t =
     { url : string
     ; default_branch : Object.resolved option Fiber.t
-    ; branches : Object.resolved String.Map.t Fiber.t
-    ; tags : Object.resolved String.Map.t Fiber.t
+    ; refs : Object.resolved String.Map.t Fiber.t
```

Comment:
Tbh, this doesn't seem like an improvement in readability. I think you could just as easily just parse the `refs/{heads,tags}` out of the query and then know which map you need to look in.

Opinions aside, I think this makes it a bit harder to implement the "did you mean" hint. That one would be rather useful when a user typos a branch or a tag. IMO you should hold off this refactoring until we have that useful hint ready.

Reply:
The problem is with other types of refs like `pull` refs for heads and merges. The dune repo itself has at least 4 types of refs and we don't even do anything particularly unusual with Git.

Git refs are basically just a `Object.resolved String.Map.t` and the string is just `refs/{heads,tags}` by convention, so to me this seems like the best way to represent them aligned with how Git treats them, thus fewer concepts to keep in mind and less ways for repos to break our assumptions.